docs/classes/response.md

```diff
@@ -3,92 +3,113 @@ title: Response
+The **`Response`** class represents an HTTP response. The **`Response`** promise is returned by [request handlers](request-handler.md) and [`middleware`](middleware.md).
 
-## `setStatus(int $code): Response`
+## Constructor
 
-Sets the numeric HTTP status code (between 100 and 599).
+```php
+public function __construct(
+    int $code = Status::OK, 
+    string[] | string[][] $headers = [], 
+    Amp\ByteStream\InputStream | string | null $stringOrStream = null
+)
+```
 
-If not assigned this value defaults to 200.
+### Parameters
 
-## `setReason(string $phrase): Response`
+|`int`|`$code`|HTTP response status code|
+|`string[]`<br />`string[][]`|`$headers`|An array of strings or an array of string arrays.|
+|[`Amp\ByteStream\InputStream`](https://amphp.org/byte-stream/)<br />`string`<br />`null`|`$stringOrStream`|Response body|
 
-Sets the optional HTTP reason phrase.
+## Destructor
 
-## `addHeader(string $field, string $value): Response`
+Invokes dispose handlers (i.e. functions that registered via [`onDispose`](#ondisposecallable-ondispose) method).
 
-Appends the specified header.
+{:.note}
+> Uncaught exceptions from the dispose handlers will be forwarded to the [event loop](https://amphp.org/amp/event-loop/) error handler.
 
-## `setHeader(string $field, string $value): Response`
+## `getBody(): Amp\ByteStream\InputStream`
 
-Sets the specified header.
+Returns the [stream](https://amphp.org/byte-stream/) for the message body.
 
-This method will replace any existing headers for the specified field.
+## `setBody(\Amp\ByteStream\InputStream | string | null $stringOrStream)`
 
-## `setCookie(string $name, string $value, array $flags = []): Response`
+Sets the [stream](https://amphp.org/byte-stream/) for the message body.
 
-Provides an easy API to set cookie headers.
+{:.note}
+> Using a string will automatically set the `Content-Length` header to the length of the given string. Setting a [`stream`](https://amphp.org/byte-stream/) will remove the `Content-Length` header.
 
-Those who prefer using addHeader() may do so.
+## `setHeader(string $name, string | string[] $value)`
 
-Valid `$flags` are per [RFC 6265](https://tools.ietf.org/html/rfc6265#section-5.2.1):
+Sets the named header to the given value.
 
-- `"Expires" => date("r", $timestamp)` - A timestamp when the cookie will become invalid (set to a date in the past to delete it)
-- `"Max-Age" => $seconds` - A number in seconds when the cookie must be expired by the client
-- `"Domain" => $domain` - The domain where the cookie is available
-- `"Path" => $path` - The path the cookie is restricted to
-- `"Secure"` - Only send this cookie to the server over TLS
-- `"HttpOnly"` - The client must hide this cookie from any scripts (e.g. Javascript)
+## `addHeader(string $name, string | string[] $value)`
 
-## `write(string $partialBodyChunk): \Amp\Promise`
+Adds the value to the named header, or creates the header with the given value if it did not exist.
```

Comment:
It doesn't add a value to any existing header, instead it simply adds a new header line with the same name if one does already exist.